lib/beaker/dsl/install_utils/windows_utils.rb

```diff
@@ -167,6 +167,13 @@ def install_msi_on(hosts, msi_path, msi_opts = {}, opts = {})
             reg_query_command = %Q(reg query "#{reg_key}" /v "RememberedPuppetAgentStartupMode" | findstr #{msi_opts['PUPPET_AGENT_STARTUP_MODE']})
             on host, Command.new(reg_query_command, [], { :cmdexe => true })
 
+            # (PA-620) environment.bat should be run before any cmd.exe shell command
+            # in order to properly set up environment variables
+            auto_run_key = "HKLM\\SOFTWARE\\Microsoft\\Command Processor\AutoRun"
+            # TODO What's the best way to get the actual path to the script, respecting arch differences?
+            environment_script = "C:\\Program Files\\Puppet Labs\\Puppet\bin\\environment.bat"
```

Comment:
This is not the correct path on all hosts. How can I correctly determine whether I will need to look in `Program Files` or `Program Files (x86)` for the script?